src/deploy/hosting/convertConfig.js

```diff
@@ -1,4 +1,25 @@
 const _ = require("lodash");
+const { FirebaseError } = require("../../error");
+
+/**
+ * makePattern contains the logic for extracting exactly one glob/regexp from
+ * a Hosting rewrite/redirect/header specification
+ */
+function makePattern(type, spec) {
+  const glob = spec.source || spec.glob;
```

Comment:
Swap the order here, `source` will become the legacy thing that gets dropped if we ever do so.

Reply:
Done.